opnode/p2p/host_test.go

```diff
@@ -0,0 +1,136 @@
+package p2p
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"net"
+	"testing"
+	"time"
+
+	ds "github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/sync"
+	"github.com/libp2p/go-libp2p-core/connmgr"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	tswarm "github.com/libp2p/go-libp2p-swarm/testing"
+	yamux "github.com/libp2p/go-libp2p-yamux"
+	lconf "github.com/libp2p/go-libp2p/config"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestingConfig(t *testing.T) *Config {
+	p, _, err := crypto.GenerateSecp256k1Key(rand.Reader)
+	assert.NoError(t, err, "failed to generate new p2p priv key")
+	mtpt, err := lconf.MuxerConstructor(yamux.DefaultTransport)
+	assert.NoError(t, err)
+	mux := lconf.MsMuxC{MuxC: mtpt, ID: "/yamux/1.0.0"}
+
+	return &Config{
+		Priv:                (*ecdsa.PrivateKey)((p).(*crypto.Secp256k1PrivateKey)),
+		DisableP2P:          false,
+		NoDiscovery:         true, // we statically peer during most tests.
+		ListenIP:            net.IP{127, 0, 0, 1},
+		ListenTCPPort:       0, // bind to any available port
+		StaticPeers:         nil,
+		HostMux:             []lconf.MsMuxC{mux},
+		NoTransportSecurity: true,
+		PeersLo:             1,
+		PeersHi:             10,
+		PeersGrace:          time.Second * 10,
+		NAT:                 false,
+		UserAgent:           "optimism-testing",
+		TimeoutNegotiation:  time.Second * 2,
+		TimeoutAccept:       time.Second * 2,
+		TimeoutDial:         time.Second * 2,
+		Store:               sync.MutexWrap(ds.NewMapDatastore()),
+		ConnGater: func(conf *Config) (connmgr.ConnectionGater, error) {
+			return tswarm.DefaultMockConnectionGater(), nil
+		},
+		ConnMngr: DefaultConnManager,
+	}
+}
+
+// Simplified p2p test, to debug/test basic libp2p things with
+func TestP2PSimple(t *testing.T) {
+	confA := TestingConfig(t)
+	confB := TestingConfig(t)
+	hostA, err := confA.Host()
+	assert.NoError(t, err, "failed to launch host A")
+	defer hostA.Close()
+	hostB, err := confB.Host()
+	assert.NoError(t, err, "failed to launch host B")
```

Comment:
assert vs require. I guess we ended up on require elsewhere. I don't have a strong opinion either way, but consistency is nice.

Reply:
I've only been using `github.com/stretchr/testify/assert` in all of my spec work so far. We also use their mock library. I think the system test uses `require` 99% to check if errors are nil with a generic `require.Nil`. And then there are like ~5 misc calls to `require`. If think we should change those others to `assert.NoError(t, err)`

Reply:
Scratch that, didn't realize `require` is from `testify` too. It just terminates the test quicker, which makes sense for error handling. I'll update my tests